src/stories/components/Footer/index.tsx

```diff
@@ -0,0 +1,30 @@
+import React, { FC, PropsWithChildren, useMemo, Children } from 'react';
+
+import Grid from '@mui/material/Grid';
+
+import * as S from './styles';
+
+export const Footer: FC<PropsWithChildren> = ({ children }) => {
+  const renderNavChildren = useMemo(() => {
+    return Children.toArray(children).slice(0, 4);
+  }, [children]);
+
+  const renderFlagsChildren = useMemo(() => {
+    return Children.toArray(children).slice(4, 8);
+  }, [children]);
+
+  const renderDisclaimerChildren = useMemo(() => {
+    return Children.toArray(children).slice(8);
+  }, [children]);
+
+  return (
+    <S.Container>
+      <S.LinkWrapper container>
+        {renderNavChildren}
+        {renderFlagsChildren}
+      </S.LinkWrapper>
+
+      <Grid item>{renderDisclaimerChildren}</Grid>
+    </S.Container>
+  );
+};
```

Comment:
Acredito que a utilização de slices no children pode deixar meio confuso para quem for utilizar.

Acredito que a solução desse footer poderia ser por passagem de props. O footer teria que determinar onde ficaria cada elemento (nav links, social media e texto).

```suggestion
type Props = {
  navLinks?: Array<{link: string; label: string}>;
  socialMedias?: Array<{link: string; label: string; icon?: InterfaceIcon}>;
  disclaimer?: string;
}

export const Footer: FC<Props> = ({ navLinks = [], disclaimer }) => {
  return (
    <S.Container>
      <S.LinkWrapper container>
        {navLinks.map(navLink => (
          <S.Link href={navLink.link}>{navLink.label}</S.Link>
        ))}
      </S.LinkWrapper>
      
      <S.SocialMediaWrapper>
        {socialMedias.map(socialMedia => (
          <S.SocialMedia href={socialMedia.link} icon={socialMedia.icon}>
            {socialMedia.label}
          </S.SocialMedia>
        ))}
      </S.SocialMediaWrapper>

      <S.Disclarimer>{disclaimer}</S.Disclaimer>
    </S.Container>
  );
};
```